data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/mapping/OpenSearchPluginAttributesMapper.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.logstash.mapping;
+
+import org.opensearch.dataprepper.logstash.exception.LogstashConfigurationException;
+import org.opensearch.dataprepper.logstash.model.LogstashAttribute;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+class OpenSearchPluginAttributesMapper extends AbstractLogstashPluginAttributesMapper {
+    protected static final String LOGSTASH_OPENSEARCH_INDEX_ATTRIBUTE_NAME = "index";
+    protected static final String REGEX_EXTRACTING_DATETIME_PATTERN = "%\\{(.*?)\\}";
+
+    @Override
+    protected void mapCustomAttributes(List<LogstashAttribute> logstashAttributes, LogstashAttributesMappings logstashAttributesMappings, Map<String, Object> pluginSettings) {
+
+        final LogstashAttribute logstashIndexAttribute = logstashAttributes.stream()
+                .filter(a -> a.getAttributeName().equals(LOGSTASH_OPENSEARCH_INDEX_ATTRIBUTE_NAME))
+                .findFirst()
+                .orElseThrow(() -> new LogstashConfigurationException("Index attribute not found"));
+
+        final String convertedIndexPatternValue = convertIndexDateTimePattern(logstashIndexAttribute);
+
+        pluginSettings.put(logstashAttributesMappings.getMappedAttributeNames().get(LOGSTASH_OPENSEARCH_INDEX_ATTRIBUTE_NAME),
+                convertedIndexPatternValue);
+
+    }
+
+    @Override
+    protected HashSet<String> getCustomMappedAttributeNames() {
+        return new HashSet<>(Collections.singletonList(LOGSTASH_OPENSEARCH_INDEX_ATTRIBUTE_NAME));
+    }
+
+    private String convertIndexDateTimePattern(LogstashAttribute logstashAttribute) {
+
+        final String logstashIndexAttributeValue = logstashAttribute.getAttributeValue().getValue().toString();
+        final Pattern pattern = Pattern.compile(REGEX_EXTRACTING_DATETIME_PATTERN);
+        final Matcher dateTimePatternMatcher = pattern.matcher(logstashIndexAttributeValue);
+
+        final StringBuilder updatedDateTimePattern = new StringBuilder();
+        String updatedIndexAttributeValue = "";
```

Comment:
As a best practice, variables should be minimally scoped to the block they are used and we should avoid re-assignment where possible. We can achieve this by leveraging an additional private method to handle the match and conversion part. 

```
return dateTimePatternMatcher.find() ? matchAndConvertDateTimePattern(dateTimePatternMatcher, logstashIndexAttributeValue) : logstashIndexAttributeValue;

```
New method:
```
private String matchAndConvertDateTimePattern(final Matcher dateTimePatternMatcher, final String logstashIndexAttributeValue) {

        final String dateTimePattern = dateTimePatternMatcher.group(0);
        final StringBuilder updatedDateTimePattern = new StringBuilder();
        for (final char character: dateTimePattern.toCharArray()) { 
           ... // conversion logic from below.
        }
        return logstashIndexAttributeValue.replace(dateTimePattern, updatedDateTimePattern);
    }
```

The current implementation also has a bug when a pattern is not provided because we are assigning the updatedIndexAttributeValue to `""` and returning `""` if a pattern does not exist.

Reply:
Thanks for catching this. It does substitute `""` for plain Strings index names. I will update this and add a test for this usecase.

Reply:
@cmanning09 A few clarifications in your suggestion:
> As a best practice, variables should be minimally scoped to the block they are used and we should avoid re-assignment where possible. We can achieve this by leveraging an additional private method to handle the match and conversion part.
> 
* When you say additional private method, did you instead mean replacing the current `convertIndexDateTimePattern(LogstashAttribute logstashAttribute)` method for `matchAndConvertDateTimePattern(final Matcher dateTimePatternMatcher, final String logstashIndexAttributeValue` method?

> ```
> return dateTimePatternMatcher.find() ? matchAndConvertDateTimePattern(dateTimePatternMatcher, logstashIndexAttributeValue) : logstashIndexAttributeValue;
> ```
* I'm not sure I understand this return statement here, can you provide more context? 
> 
> New method:
> 
> ```
> private String matchAndConvertDateTimePattern(final Matcher dateTimePatternMatcher, final String logstashIndexAttributeValue) {
> 
>         final String dateTimePattern = dateTimePatternMatcher.group(0);
>         final StringBuilder updatedDateTimePattern = new StringBuilder();
>         for (final char character: dateTimePattern.toCharArray()) { 
>            ... // conversion logic from below.
>         }
>         return logstashIndexAttributeValue.replace(dateTimePattern, updatedDateTimePattern);
>     }
> ```
Based on the code snippet provided for the new method, are we okay with defining `Matcher` in the `mapCustomAttributes` method instead of the convert method?
> 
> The current implementation also has a bug when a pattern is not provided because we are assigning the updatedIndexAttributeValue to `""` and returning `""` if a pattern does not exist.